src/CommunityToolkit.Maui/Interfaces/IPopupService.shared.cs

```diff
@@ -0,0 +1,56 @@
+﻿using CommunityToolkit.Maui.Core;
+
+namespace CommunityToolkit.Maui;
+
+/// <summary>
+/// Provides a mechanism for displaying Popups based on the underlying view model.
+/// </summary>
+public interface IPopupService
```

Comment:
We have lost the ability to pass data across to the view model when it is being presented (the old `onPresenting` parameter). Do we have an alternative here or is this another breaking change to document?

Reply:
Could you explain more about what you mean? 

When the developer calls `Popup.Close()` or `PopupService.Close()` in the ViewModel, they give us the result. 

This result is then returned to the code that is await'ing `Popup.ShowAsync()` (likely the View).

Reply:
If the user calls the Popup from the View, they don't need to pass in a result that we'll pass directly back to them.

Reply:
To be honest, my vote is to remove `Popup<T>` completely because it just turns Popup into a Messaging Service which is entirely unnecessary. There already exists a million ways for developers to propagate data between the View and ViewModel.

Reply:
I'm actually talking about the other direction here. Setting values on the view model behind the popup, a bit like what can be done with IQueryAttributable in shell.

Reply:
I'm still not sure I understand. 

To change a value in the ViewModel behind the popup, the developer would use bindings like they normally would.

Is there a real-world example that concerns you specifically?

Reply:
What has prompted this conversation is this section of the current documentation which would need to be deleted: https://learn.microsoft.com/dotnet/communitytoolkit/maui/views/popup-service#passing-data-to-a-popup-view-model

By change a value I mean perhaps set a message in a popup that you want to show to the user. Not something that needs to be bound but needs to be passed across to the underlying view model for the popup. Does that make sense?

Reply:
Also this isn't reinventing the wheel... let me try to explain:

We have a view model `ExamResultsViewModel` which shows exam result for a student. When the user taps on a result they can see some more information in a popup, let's call it `ExamResultsDetailPopup` which has a backing view model called `ExamResuktsDetailsPopupViewModel`). The scenario I'm talking about wants to get the data from `ExamResultsViewModel` to `ExamResuktsDetailsPopupViewModel`. 

Using messaging would be a messy because you want to get the data to the view model before the popup is displayed. So this scenario is trying to match what is available with `IQueryAttributable` when navigating between pages in Maui

Reply:
Ah - ViewModel to ViewModel. Yea, that makes sense. Got it. 

Yea, let's still move forward with this release and add that in if we get feedback from the community, similar to Anchor. 

I do wonder (not for this PR) if we could implement `IQueryable` for Popup so that it "just works".